project/yxuan0329/proposal.rst

```diff
@@ -0,0 +1,171 @@
+=================
+R-Tree Spatio-Temporal Search Engine
+=================
+
+.. This is a template to specify what your project is and the execution plan.  You
+.. will find it is difficult to plan for things that you are not sure about.  Do
+.. your best.
+
+.. note::
+
+..   You are encouraged to use this plain-text `reStructuredText
+..   <https://docutils.sourceforge.io/rst.html>`__ format.  :download:`Download
+..   this file. <project_template.rst>`
+
+.. Please do give your project a name that clearly states the subject and is short
+.. enough for people to remember.
+
+.. Additional documents or files may be used.  They should be added in the
+.. repository.
+
+Basic Information
+=================
+
+.. Create a GitHub repository to host your project and add the URL here.  The
+.. GitHub repository has an 'About' field, in which you are encouraged to write a
+.. simple statement (preferably one sentence) to introduce the project.
+
+R-tree is a tree structure designed for efficient spatial access, it has been 
+widely used in geographic searching. However, when searching for data in map 
+system, it is crucial to consider temporal features to enhance searching 
+performance. For instance, when searching for "food" in the morning, users 
+would probably expect to receive the breakfast places that are open at that 
+time, rather than the closed breakfast places or some other unrelated lunch 
+restaurants or bars. Therefore, a search engine that can box the nearest 
+distance data and arrange the results based on the temporal information may be
+useful for the users.
+
+The R-Tree Spatio-Temporal Search Engine aims to provide a search engine for 
+users to query multi-dimensional data with both spatial and temporal 
+information. In this project, we will provide a demonstration of the R-tree 
+search engine, enabling users to enter queries and receive a set of results 
+that include the nearest distance and the best time-matched entities.
+
+
+Problem to Solve
+================
+
+.. Describe the problem or the set of problems you want to solve.  Include
+.. necessary background information without making it lengthy.
+
+.. Some points may help you organize the problem description:
+
+.. 1. The field or industry of the problem.
+.. 2. The physics and/or the mathematics behind the problem.
+.. 3. The algorithm or numerical method that should be applied for solving the
+..    problem.
+
+R-tree is a data structure renowned for its exceptional performance in 
+searching multi-dimensional data, it plays a pivotal role in various 
+applications, particularly in spatial information systems such as maps. 
+However, beyond spatial information, data in map system should also take 
+temporal feature in considerations for better searching performance. When 
+searching nearest location on map system, users should be able to see different 
+sorting results depending on their specified searching time. In other words, 
```

Comment:
It sounds like to need a database system to get different results at different times.  It may not be just a spatial index like R-tree.